src/libostree/ostree-repo-private.h

```diff
@@ -363,4 +364,16 @@ _ostree_repo_get_remote_inherited (OstreeRepo  *self,
                                    const char  *name,
                                    GError     **error);
 
+gboolean
+_ostree_repo_lock_shared (OstreeRepo        *self,
+                          gboolean           non_blocking,
+                          GLnxLockFile      *lockfile,
+                          GError           **error);
+
+gboolean
+_ostree_repo_lock_exclusive (OstreeRepo        *self,
```

Comment:
do you think it would make sense to expose these two functions externally?  In general I think they can be useful when more complex prune logics are implemented by an application.  In the particular case, there is still an issue with Atomic system containers when `atomic images prune` is used at the same time an image referring to a dangling layer (which is already present locally) is being pulled: prune might delete the layer but the pull process already checked it is present in the repository and it doesn't pull it again.  If these two functions are made public, I could use them directly, otherwise we will probably need to implement the locking part separately.